packages/router/src/lib/operations/cancel.ts

```diff
@@ -96,17 +102,65 @@ export const cancel = async (
     cancelChain = invariantData.receivingChainId;
   }
 
-  // Send to tx service
-  logger.info("Sending cancel tx", requestContext, methodContext, { side });
-
-  const receipt = await contractWriter.cancel(
-    cancelChain,
-    {
-      txData: { ...invariantData, ...variant },
-      signature: "0x",
-    },
-    requestContext,
-  );
+  let receipt: providers.TransactionReceipt;
+
+  if (isRouterContract) {
+    routerRelayerFeeAsset = utils.getAddress(
+      config.chainConfig[invariantData.receivingChainId].routerContractRelayerAsset || AddressZero,
+    );
+    const relayerFeeAssetDecimal = await txService.getDecimalsForAsset(
+      invariantData.receivingChainId,
+      invariantData.receivingAssetId,
+    );
```

Comment:
isn't this the wrong asset?

Reply:
Oops, this needs to depend on the chain they want to cancel on cc @sanchaymittal!